### DISCUSSION TITLE
How to see connection time and read time?

### DISCUSSION THREAD

**Discussion by talhatek** [encode/httpx#3254] [Q&A] [answered]
Hello everyone i wanna analysis my network calls. I can estimate the entire request time but i wanna see my avg connection times and read times. Is there a way for that? Thx.

**Comment by karpetrosyan** [comment, marked as answer]
Hi! Take a look at [this pr](https://github.com/encode/httpcore/pull/932).
I think we can do this using the traces called `connect_tcp` and `read`.

@tomchristie fyi

**Comment by talhatek** [comment]
I dont see any read event on https://www.python-httpx.org/advanced/extensions/ but i can estimate roughly. thanks

**Reply by karpetrosyan** [reply]
Sorry, it's called `receive_response_body `, see https://www.python-httpx.org/advanced/extensions/#trace
